File: python/3263_convert_doubly_linked_list_to_array_i.py
# Problem 3263: Convert Doubly Linked List to Array I
# https://leetcode.com/problems/convert-doubly-linked-list-to-array-i/

from typing import List, Optional


# Definition for a Node.
class Node:
    def __init__(self, val, prev=None, next=None):
        self.val = val
        self.prev = prev
        self.next = next

class Solution:
    def to_array(self, root: Optional[Node]) -> List[int]:
        # 開空的 list 跑迴圈把 node 的值一個一個加在後面
        ans = []
        while root:
            ans.append(root.val)
            root = root.next
        return ans
        

File: c++/3263_ConvertDoublyLinkedListToArrayI.cpp
// Problem 3263: Convert Doubly Linked List to Array I
// https://leetcode.com/problems/convert-doubly-linked-list-to-array-i/

#include <vector>
using namespace std;

// Definition for doubly-linked list.
class Node {
public:
    int val;
    Node* prev;
    Node* next;
    Node() : val(0), next(nullptr), prev(nullptr) {}
    Node(int x) : val(x), next(nullptr), prev(nullptr) {}
    Node(int x, Node *prev, Node *next) : val(x), next(next), prev(prev) {}
};

class Solution {
public:
	vector<int> toArray(Node *head){
        vector<int> ans;
        while (head) {
            ans.push_back(head -> val);
            head = head -> next;
        }
        return ans;
    }
};